lib/src/testing/fake_platform.dart

```diff
@@ -13,22 +14,22 @@ class FakePlatform extends Platform {
   /// Unspecified properties will *not* be assigned default values (they will
   /// remain `null`).
   FakePlatform({
-    this.numberOfProcessors,
-    this.pathSeparator,
-    this.operatingSystem,
-    this.operatingSystemVersion,
-    this.localHostname,
-    this.environment,
-    this.executable,
-    this.resolvedExecutable,
-    this.script,
-    this.executableArguments,
+    required this.numberOfProcessors,
```

Comment:
Making these required is going to be a big breaking change since most uses of `FakePlatform` are when you know in a test that you only need to fake one field.

Maybe make private fields that are nullable, then the getters that you override for the interface either return the field if it's set, or throw?

Reply:
Yeah, that sounds like a good idea.

Reply:
Updated